src/puppetlabs/puppetdb/command.clj

```diff
@@ -249,6 +272,16 @@
                id (command-names :store-report)
                puppet_version certname)))
 
+(defmethod process-command! [(command-names :store-report) 3]
+  [{:keys [version] :as command} {:keys [db]}]
+  (store-report* 5 db (-> command
+                          (update :payload report/wire-v3->wire-v5 (get-in command [:annotations :received])))))
```

Comment:
For reports we have the conversion in the `report` namespace, but for `replace-facts` is inline. Seems like it should be one or the other.